packages/core/src/core/adapter_format.ts

```diff
@@ -56,13 +65,73 @@ const getAdapterAndContext = async ({
     ignoreStateElemIdMappingForSelectors,
   })
   const adapterContext = adaptersCreatorConfigs[accountName]
-  const adapter = adapterCreators[adapterName]
   return { adapter, adapterContext, resolvedElements }
 }
 
+type IsInitializedFolderArgs = {
+  baseDir: string
+} & GetAdapterArgs
+
+export type IsInitializedFolderResult = {
+  result: boolean
+  errors: ReadonlyArray<SaltoError>
+}
+
+export const isInitializedFolder = async ({
+  baseDir,
+  workspace,
+  accountName,
+}: IsInitializedFolderArgs): Promise<IsInitializedFolderResult> => {
+  const { adapter } = getAdapter({ workspace, accountName })
+  if (adapter.adapterFormat === undefined || adapter.adapterFormat.isInitializedFolder === undefined) {
+    return {
+      result: false,
+      errors: [
+        {
+          severity: 'Error' as const,
+          message: 'Format not supported',
+          detailedMessage: `Account ${accountName}'s adapter does not support checking a non-nacl format folder`,
+        },
+      ],
+    }
+  }
+
+  return {
+    result: await adapter.adapterFormat.isInitializedFolder({ baseDir }),
+    errors: [],
+  }
```

Comment:
readability nit - putting the main operation here nested inside complex construction of the return value makes it harder to notice.
```suggestion
  const result = await adapter.adapterFormat.isInitializedFolder({ baseDir })
  return { result , errors: [] }
```

Reply:
also, should this be in a `try...catch`?